### DISCUSSION TITLE
Системные таблицы после обновления версии.

### DISCUSSION THREAD

**Discussion by vldekov** [totumonline/totum-ru-issues-and-discussions#465] [Как это сделать?] [answered]
Если добавить свои дополнительные поля в системную таблицу USERS, то после обновления версии эти поля останутся?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Обновление (которое кстати работает на том же механизме что импорт-экспорт) только обновляет и добавляет поля.

Поэтому если вы добавили поле — то оно останется.

Если вы модифицировали поле — то оно перезапишется. Для защиты от перезаписи в таких случаях в PRO есть Удерживаемые поля https://ru.docs.totum.online/update#held-fields. В этом случае при обновлении в консоли будут показаны те поля, которые были удержаны при обновлении. Важно: удержание системных полей может привести к неработоспособности системы после обновления.

В каком случае может быть такое, что вы добавили, а оно перезаписалось — если в обновлении придет поле в этой таблице с таким-же name. Поэтому когда мы добавляем новые системные поля в существующие таблицы их name всегда начинаются с `ttm__`

**Reply by ttm-alexeypolunin** [reply]
Не рекомендуется в таблице users добавлять много тк строка из нее читается при любом действии в Тотум тк проверка прав происходит при любом действии.

Сейчас идет работа над блоком, который в PRO будет делать копию users в оперативку `Redis`, что ускорит это место для систем с большим количеством пользователей, но все-равно не стоит там писать четырехэтажную логику.
